src/mobile/src/ui/components/EnterPassword.js

```diff
@@ -103,7 +105,9 @@ class EnterPassword extends Component {
     }
 
     hideModal() {
-        this.props.toggleModalActivity();
+        if (this.props.isModalActive) {
+            this.props.toggleModalActivity();
```

Comment:
What happens if `toggleModalActivity` is dispatched when modal is already inactive?

Reply:
Well this function is just for hiding the modal.

Reply:
Oh didn't realise it was for just hiding modal as it also [accepts](https://github.com/iotaledger/trinity-wallet/blob/develop/src/shared/actions/ui.js#L217) `modalProps` and `modalContent`. 

Perhaps the condition (if modal is active) is moved to the redux action itself. 

```
const hideModal = () => {
    return (dispatch, getState) => {
        if (getState().ui.isModalActive) { dispatch(toggleModalActivity()); };
    };
};
```

That way we won't need to include `isModalActive` prop in every component where we need to close the modal.

Reply:
This was more as a failsafe. I don't think they are necessary however - I will remove them.

Reply:
FYI, toggleModalActivity has two purposes - opening the right modal with the right props. but without props it simply closes an open modal.